build/localnet/README.md

```diff
@@ -0,0 +1,165 @@
+
+### Interacting w/ LocalNet
+
+As the workloads run in Kubernetes, you can see and modify any resources on your local kubernetes by a tool of your choice (k9s, Lens, VSCode extension, etc), but note that Tilt will change manifests back eventually.
+
+#### Make Targets
+
+Open a shell in the pod that has `client` cli available. It gets updated automatically whenever the code changes:
+
+```bash
+make localnet_shell
+```
+
+Open a `client debug` cli. It allows to interact with blockchain, e.g. change pace maker mode, reset to genesis, etc. It gets updated automatically whenever the code changes (though you would need to stop/start the binary to execute the new code):
+
+```bash
+make localnet_client_debug
+```
+
+## How does it work?
+
+[tilt](https://tilt.dev/) reads the [`Tiltfile`](../../Tiltfile), where LocalNet configs are specified. `Tiltfile` is written in [Starlark](https://github.com/bazelbuild/starlark), a dialect of Python.
+
+The k8s manifests that `tilt` submits to the cluster can be found in [this directory](./):
+
+- **[dependencies](./dependencies/)**: a helm chart that installs all necessary dependencies to run and observe LocalNet (postgresql, prometheus, grafana, etc).
+- **[4 Validators](./v1-validator-template.sh)**: The validator binary that runs inside of the container gets updated automatically and process restarted on each code change (i.e. hot reloads).
+- **[V1 CLI client](./cli-client.yaml)**: This binary that can be used to perform debug operations. Run `make localnet_client_debug` to execute commands such as `ResetToGenesis` or `TogglePacemakerMode`. This binary is also automatically updated when you make changes to the codebase.
+
+Tilt continuously monitors files on local filesystem, and it rebuilds the binary and distributes it to the pods on every code change. This allows developers to iterate on the code and see the changes immediately.
+
+## Troubleshooting
+
+### Why?
+
+Developers might experience issues with running LocalNet on Kubernetes.
+
+Issues might be related to the fact different developers run different clusters/OS/environments.
+
+Machines going to sleep and that might not play well with virtual machines, postgres or pocket client.
+
+### Force Trigger an Update
+
+Visit the tilt web UI by pressing `space` in the shell where you started tilt or by visiting [this webpage](http://localhost:10350/).
+
+If you see any errors, you can click `Trigger Update` on a resource that has issues to restart the service or retry a command.
+
+### Force Restart
+
+If force triggering an update didn't work, try the following:
+
+1. `make localnet_down`
+2. `make localnet_up`
+
+### Full Cleanup
+
+If a force restart didn't help, try rebuilding local kubernetes cluster using the tool you're managing your cluster with (e.g. Docker Desktop, Rancher Desktop, k3s, k3d, minikube, etc).
+
+### Docker Desktop
+
+# TODO_IN_THIS_COMMIT
```

Comment:
Can you please update this based on our offline conversation?

Reply:
What exactly do you want to see here? Docker Desktop, as any other software, might not work (frankly, I was never happy with it on mac). If you don't want to troubleshoot issues, you can resort to other software.

In this readme file, I included a list of distributions I've tested this LocalNet on, linked a page on tilt.dev website describing all possible options (basically - any cluster/distribution will work, as long as Kubernetes is available).